docs/MockFunctions.md

```diff
@@ -127,6 +127,42 @@ dependent component and configuring that, but the technique is the same. In
 these cases, try to avoid the temptation to implement logic inside of any
 function that's not directly being tested.
 
+## Mocking Modules
+
+Suppose we have a class that fetches users from our API. The class uses [axios](https://github.com/axios/axios) to call the API then returns the `data` attribute which contains all the users:
+
+```js
+// users.js
+import axios from 'axios'
+
+class Users {
+  static all() {
+    return axios.get('/users.json').then(resp => resp.data)
+  }
+}
+
+export default Users
+```
+
+Now, in order to test this method without actually hitting the API (and thus creating slow and fragile tests), we can use the `jest.mock(...)` function to automatically mock the axios module.
+
+Once we mock the module we can spy on the `.get` method and provide a `mockImplementation` that returns the data we want our test to assert against. In effect, we are saying that we want axios.get('/users.json') to return a fake response.
+
+```js
+// users.test.js
+import axios from 'axios'
+import Users from './users'
+
+jest.mock('axios')
+
+test('should fetch users', () => {
+  const resp = Promise.resolve({ data: [{ name: 'Bob' }] })
+  jest.spyOn(axios, 'get').mockImplementationOnce(() => resp)
```

Comment:
wouldn't this be

`axios.get.mockReturnValue(resp);`

since it is already mocked..?

Reply:
Nice, even simpler!